drivers/platform/pico/pico_i2c.c

```diff
@@ -74,7 +96,7 @@ int32_t pico_i2c_init(struct no_os_i2c_desc **desc,
 	pico_i2c = (struct pico_i2c_desc *)calloc(1,sizeof(*pico_i2c));
 	if (!pico_i2c) {
 		ret = -ENOMEM;
-		goto error;
+		goto free_desc;
```

Comment:
I believed this change is unrelated to the commit subject.

Reply:
Thank you! I added another commit with these changes.